crates/bevy_ecs/src/storage/sorted_small_vec.rs

```diff
@@ -0,0 +1,259 @@
+use smallvec::SmallVec;
+use core::cmp::Ordering;
+
+/// Stores a sorted list of indices with quick implementation for union, difference, intersection.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct SortedSmallVec<const N: usize>(SmallVec<[usize; N]>);
```

Comment:
I thought about it and decided against it. It isn't needed now and I don't think it'll be needed in the near future. If anyone were to need a generalized version later, it would be easy to implement.

Right now it seems like overengineering.